block.go

```diff
@@ -0,0 +1,42 @@
+package slack
+
+// @NOTE: Blocks are in beta and subject to change.
+
+// More Information: https://api.slack.com/block-kit
+
+var (
+	// validBlockList contains a list of
+	validBlockList = []string{
+		"section",
+		"divider",
+		"image",
+		"actions",
+		"context",
+	}
+)
+
+// Block defines an interface all block types should implement
+// to ensure consistency between blocks.
+type Block interface {
```

Comment:
interface is useless, method doesn't really provide anything to the api.

i'd remove the method and make this a private interface. this will

```
type block interface{
  blockType() MessageBlockType
}
```

this will give us the ability to enforce specific constraints on the input to `NewBlock` without exposing any details in the public api.